internal/pkg/argocd/argocd.go

```diff
@@ -170,26 +172,90 @@ func generateDiffOfAComponent(ctx context.Context, componentPath string, prBranc
 	}
 	foundApps, err := appIf.List(ctx, &appLabelQuery)
 	if err != nil {
-		log.Errorf("Error listing ArgoCD applications: %v", err)
-		componentDiffResult.DiffError = err
-		return componentDiffResult
+		return nil, fmt.Errorf("Error listing ArgoCD applications: %v", err)
+
 	}
 	if len(foundApps.Items) == 0 {
-		componentDiffResult.DiffError = fmt.Errorf("No ArgoCD application found for component path %s(repo %s), used this label selector: %s", componentPath, repo, labelSelector)
+		return nil, fmt.Errorf("No ArgoCD application found for component path sha1 %s(repo %s), used this label selector: %s", componentPathSha1, repo, labelSelector)
+	}
+
+	// we expect only one app with this label and repo selectors
+	return &foundApps.Items[0], nil
+}
+
+// findArgocdAppByManifestPathAnnotation is the default method to find an ArgoCD application by the manifest-generate-paths annotation.
+// It assumes the ArgoCD (optional) manifest-generate-paths annotation is set on all relevant apps.
+// Notice that this method includes a full list of all ArgoCD applications in the repo, this could be a performance issue if there are many apps in the repo.
+func findArgocdAppByManifestPathAnnotation(ctx context.Context, componentPath string, repo string, appIf application.ApplicationServiceClient) (app *argoappv1.Application, err error) {
+	// argocd.argoproj.io/manifest-generate-paths
+	appQuery := application.ApplicationQuery{
+		Repo: &repo,
+	}
+	// AFAIKT I can't use standard grpc instrumentation here, since the argocd client abstracts too much (including the choice between Grpc and Grpc-web)
+	// I'll just manually log the time it takes to get the apps for now
+	getAppsStart := time.Now()
+	allRepoApps, err := appIf.List(ctx, &appQuery)
```

Comment:
I couldn't find it in the client library, but I guess there is no pagination here 🤔 !

Reply:
The client doesn't expose any related config (AFAIKT) so I assumed this just handled by some gRPC implantation magic

Reply:
in gRPC a request and response could be comprised of multiple "messages" (streaming) so I kind of assumed pagination support is implied